pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -470,6 +471,12 @@ func ValidateWorkerKubernetesVersionSkew(new, old *Cluster) field.ErrorList {
 	}
 	for _, nodeGroupNewSpec := range new.Spec.WorkerNodeGroupConfigurations {
 		newVersion := nodeGroupNewSpec.KubernetesVersion
+
+		if newVersion != nil && nodeGroupNewSpec.MachineGroupRef.Kind == TinkerbellMachineConfigKind {
```

Comment:
Do we have this block for CLI already or is it not needed there?

Reply:
although it's located in the webhook, this function is used in the cli as well

Reply:
Let's create an issue to move this out then as a followup.